bestla/bestla/bestla_device.h

```diff
@@ -303,10 +303,22 @@ class CpuDevice {
           for (auto& i : SMT_core) printf("%d,", i);
           printf("\n");
         }
-        E_L1Cache = L1[E_core[0]];
-        E_L2Cache = L2[E_core[0]] / 4;
-        L1Cache = E_L1Cache > L1[P_core[0]] / 2 ? L1[P_core[0]] / 2 : E_L1Cache;
-        L2Cache = E_L2Cache > L2[P_core[0]] / 2 ? L2[P_core[0]] / 2 : E_L2Cache;
+        if (!E_core.empty() && !P_core.empty()) {
+          E_L1Cache = L1[E_core[0]];
+          E_L2Cache = L2[E_core[0]] / 4;
+          int scale = SMT_core.empty() ? 1 : 2;
+          L1Cache = E_L1Cache > L1[P_core[0]] / scale ? L1[P_core[0]] / scale : E_L1Cache;
+          L2Cache = E_L2Cache > L2[P_core[0]] / scale ? L2[P_core[0]] / scale : E_L2Cache;
+        } else if (!P_core.empty()) {
+          int scale = SMT_core.empty() ? 1 : 2;
+          L1Cache = L1[P_core[0]] / scale;
```

Comment:
L1 is uint32, can you divide it directly?

Reply:
fix it

Reply:
l1 cache is counted by byte, no need to worry